docs/source/quilt_proposal.md

```diff
@@ -0,0 +1,324 @@
+additional instructions for phase).
+
+Here's a table explaining the differences between these three values that are
+tracked through the program:
+
+| Name      | Initial Value                     | Valid Values          | Can be parameterized? |
+|-----------|-----------------------------------|-----------------------|-----------------------|
+| Frequency | INITIAL-FREQUENCY in DEFFRAME     | Real numbers          | Yes                   |
+| Phase     | 0.0                               | Real numbers          | Yes                   |
+| Scale     | 1.0                               | Real numbers          | Yes                   |
+
+
+### Pulses
+
+Now that frequency, phase, and scale on a frame have been established we can
+play pulses. Pulses can be played by using the `PULSE` instruction and
+specifying both the qubit frame as well as the waveform.
+
+Given a waveform `my_custom_waveform` and the following program:
+```
+SET-FREQUENCY 0 "xy" 5400e6
+SET-PHASE 0 "xy" pi/2
+SET-SCALE 0 "xy" 1/2
+PULSE 0 "xy" my_custom_waveform
+```
+A compiler would have several options depending on the hardware backend. It
+could create a new waveform (eg. `my_custom_waveform_2`) and apply the
+(frequency, phase, scale) to it. Or it could take advantage of built-in hardware
+instructions to apply those values internally. This would be the responsibility
+of the compiler to make a trade-off between number of instructions and number of
+waveforms, given some hardware constraints.
+
+### Readout
+
+To readout a qubit the capture instruction is used. It takes a qubit frame, a
+waveform, and a classical memory reference. In this case the waveform is used
+as an integration kernel. The inner product of the integration kernel and the
+list of measured IQ values is evaluated to produce a single complex result.
+
+This complex number needs to be stored in Quil classical memory. Quil does not
+currently support complex typed memory regions, so a real array of length 2 is used instead:
+```
+# Simple capture of an IQ point
+DECLARE iq REAL[2]
+CAPTURE 0 "ro" boxcar_kernel(duration: 1e-6) iq
+```
+
+### Timing
+
+Analog control instructions introduce the concept of time into Quil. In this new interpretation, each instruction in
+Quil has an associated execution duration (which may be effectively zero for
+certain operations). It is up to the translator/hardware provider to provide semantics for how
+pulse and capture operations are scheduled, as long as certain consistency requirements are met.
+In general:
+
+1. "Events" on a frame happen at a well defined time since eg. updating a
+   frame frequency means that it starts to accumulate phase at a new rate.
+2. Events happen in the order listed in the program.
+3. Pulses on a common frame may not overlap in time.
+4. Pulses on distinct frames which involve a common qubit may not overlap in
+   time unless one is marked as `NONBLOCKING`.
+   
+A more precise specification of the timing semantics is discussed TODO
```

Comment:
- [ ] TODO

Reply:
Meant to reference this: https://github.com/rigetti/quil/blob/master/rfcs/analog/scheduling.md

This is not yet supported in the translator yet, though. This spec is less well-settled than the quilt overall proposal and should be left for comment and to mature before we promise that we hold to it.